### PR TITLE
Version linter for each mainline branch

### PR DESCRIPTION
**Pull request by dkijania**
Introduce 3 steps for version linter job. Each one guarded by `if `build.pull_request.base_branch .... condition

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

File: buildkite/src/Jobs/Test/VersionLint.dhall [hunk 1]
```diff
@@ -35,8 +35,10 @@ let buildTestCmd
                 # RunInToolchain.runInToolchain
                     ([] : List Text)
                     "buildkite/scripts/version-linter.sh ${release_branch}"
-            , label = "Versioned type linter"
-            , key = "version-linter"
+            , label = "Versioned type linter for ${release_branch}"
+            , key = "version-linter-${release_branch}"
+            , if = Some
```
Comment (mrmr1993) [inline buildkite/src/Jobs/Test/VersionLint.dhall:40]: Can you remove this and make them all soft-fail? I'll update the branch protection rules to make them required on the relevant branches then.

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

File: buildkite/src/Jobs/Test/VersionLint.dhall (hunk 1 above)
Comment (Brycewetzel) [reply buildkite/src/Jobs/Test/VersionLint.dhall]: yeah do that please.

**Comment by dkijania** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Review by SanabriaRusso — APPROVED** [review]

**Pull request merged**